source/include/core_http_client.h

```diff
@@ -708,6 +710,12 @@ HTTPStatus_t HTTPClient_AddRangeHeader( HTTPRequestHeaders_t * pRequestHeaders,
  * #HTTP_MAX_CONTENT_LENGTH_HEADER_LENGTH for the maximum Content-Length header
  * field and value that could be written to the buffer.
  *
+ * By default HTTP_FLUSH_NETWORK_SOCKET_ON_ERROR is set to 1. Therefore, upon
+ * any error in parsing the response, the network socket is flushed using the
```

Comment:
```suggestion
 * any error in parsing the response, the network socket is flushed of any remaining data using the
```

Reply:
Thanks for your review. I think I may be closing this PR and adding the logic with the send and recv timeout.